tika-parsers/src/main/resources/org/apache/tika/parser/captioning/tf/model_wrapper.py

```diff
@@ -0,0 +1,347 @@
+        Oriol Vinyals, Alexander Toshev, Samy Bengio, Dumitru Erhan
+
+        For more details, please visit : http://arxiv.org/abs/1411.4555
+    """
+
+    def __init__(self):
+
+        # scale used to initialize model variables
+        self.initializer_scale = 0.08
+
+        # dimensions of Inception v3 input images
+        self.image_height = 299
+        self.image_width = 299
+
+        # image format ("jpeg" or "png")
+        self.image_format = "jpeg"
+
+        # LSTM input and output dimensionality, respectively
+        self.embedding_size = 512
+        self.num_lstm_units = 512
+
+        # number of unique words in the vocab (plus 1, for <UNK>)
+        # the default value is larger than the expected actual vocab size to allow
+        # for differences between tokenizer versions used in preprocessing, there is
+        # no harm in using a value greater than the actual vocab size, but using a
+        # value less than the actual vocab size will result in an error
+        self.vocab_size = 12000
+
+        # reader for the input data
+        self.reader = tf.TFRecordReader()
+
+        # to match the "Show and Tell" paper we initialize all variables with a
+        # random uniform initializer
+        self.initializer = tf.random_uniform_initializer(
+            minval=-self.initializer_scale,
+            maxval=self.initializer_scale)
+
+        # a float32 Tensor with shape [batch_size, height, width, channels]
+        self.images = None
+
+        # an int32 Tensor with shape [batch_size, padded_length]
+        self.input_seqs = None
+
+        # an int32 Tensor with shape [batch_size, padded_length]
+        self.target_seqs = None
+
+        # an int32 0/1 Tensor with shape [batch_size, padded_length]
+        self.input_mask = None
+
+        # a float32 Tensor with shape [batch_size, embedding_size]
+        self.image_embeddings = None
+
+        # a float32 Tensor with shape [batch_size, padded_length, embedding_size]
+        self.seq_embeddings = None
+
+        # collection of variables from the inception submodel
+        self.inception_variables = []
+
+        # global step Tensor
+        self.global_step = None
+
+    def process_image(self, encoded_image, resize_height=346, resize_width=346, thread_id=0):
+        """Decodes and processes an image string"""
+
+        # helper function to log an image summary to the visualizer. Summaries are
+        # only logged in thread 0
+        def image_summary(name, img):
+            if not thread_id:
+                tf.summary.image(name, tf.expand_dims(img, 0))
+
+        # decode image into a float32 Tensor of shape [?, ?, 3] with values in [0, 1)
+        with tf.name_scope("decode", values=[encoded_image]):
+            if self.image_format == "jpeg":
+                image = tf.image.decode_jpeg(encoded_image, channels=3)
+            elif self.image_format == "png":
+                image = tf.image.decode_png(encoded_image, channels=3)
```

Comment:
very nice :+1: TODO: We should port this PNG decoding code to Inception-V4 based object recognition API too